File: programmers/문자열 압축.py
def solution(s):
    answer = 9999
    # print(len(s))

    for i in range(1, len(s)//2+1):

        substrings = [s[:i]]
        counts = [1]

        for j in range(1, len(s)//i+1):
            part = s[i*j:i*(j+1)]
            if part == substrings[-1]:
                counts[-1] += 1
            else:
                substrings.append(part)
                counts.append(1)

        print(substrings)
        print(counts)

        length = 0
        for k in range(len(counts)):
            length += len(substrings[k])

            if counts[k] == 1:
                continue
            else:
                length += len(str(counts[k]))

        # print(length)
        # print()
        if length < answer:
            answer = length

    return answer


print(solution("a"))

File: _programmers/문자열 압축.js
function splitter(str, n) {
    var result = [];
    var k;
    if (str.length % n) {
        k = str.length / n;
    } else {
        k = Math.ceil(str.length / n);
    }

    for (let j = 0; j < k; j++) {
        result.push(str.slice(n * j, n * (j + 1)));
    }
    return result;
}


function solution(s) {
    var answer = s.length;
    var len = 0;

    for (let i = 1; i < s.length / 2 + 1; i++) {
        let splits = splitter(s, i);
        let strings = [splits[0]], counts = [1];

        for (let j = 1; j < splits.length; j++) {
            if (splits[j] === strings[j - 1]) {
                strings[j - 1] = '';
                strings.push(splits[j]);
                counts.push(counts[j - 1] + 1);
                counts[j - 1] = '';
            } else if (counts[j - 1] === 1) {
                strings.push(splits[j]);
                counts.push(1);
                counts[j - 1] = '';
            } else {
                strings.push(splits[j]);
                counts.push(1);
            }
        }
        if (counts[counts.length - 1] === 1) counts[counts.length - 1] = '';
        counts.push('');

        len = strings.join('').length + counts.join('').length;

        if (len < answer) {
            answer = len;
        }
    }
    console.log(answer);

    return answer;
}


solution("abcabcdede")
